runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4758,10 +4763,33 @@ J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator
 
    bool use64BitClasses = comp->target().is64Bit() && !TR::Compiler->om.generateCompressedObjectHeaders();
 
-   // Init class field, then jump to end of ICF
    generateRXInstruction(cg, use64BitClasses ? TR::InstOpCode::STG : TR::InstOpCode::ST, node, classReg, generateS390MemoryReference(targetReg, TR::Compiler->om.offsetOfObjectVftField(), cg));
-   cursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
-   iComment("Init class field and jump to end of ICF.");
+#if defined(TR_TARGET_64BIT)
+   if (isIndexableDataAddrPresent)
```

Comment:
Just a question, is this the most appropriate location for the Assert (I believe it is not code-gen or this evaluator specific, right ? and have same FATAL assert in other code-gen as well)

Reply:
That's where we rely on the assumption about dataAddr field offset. I don't think we make that assumption anywhere else besides when inlining allocation of arrays (in `multianewArrayEvaluator`, `VMnewEvaluator` and couple of related helpers). So having the asserts there would make the broken code easy to find, but I can definitely move the asserts if there's a more appropriate location.

Reply:
That is ok, I just asked